subsys/net/lib/conn_mgr/conn_mgr.c

```diff
@@ -191,6 +209,127 @@ K_THREAD_DEFINE(conn_mgr, CONFIG_NET_CONNECTION_MANAGER_STACK_SIZE,
 		(k_thread_entry_t)conn_mgr_handler, NULL, NULL, NULL,
 		THREAD_PRIORITY, 0, 0);
 
+static void conn_mgr_all_if_up_cb(struct net_if *iface, void *user_data)
+{
+	int *first_status = (int *)user_data;
+	int status = net_if_up(iface);
+
+	if (status == 0) {
+		return;
+	}
+
+	if (*first_status == 0) {
+		*first_status = status;
+	}
+
+	NET_ERR("net_if_up failed for Iface %d (%p). ERR: %d",
+		net_if_get_by_iface(iface), iface, status);
+}
+
+static void conn_mgr_all_if_down_cb(struct net_if *iface, void *user_data)
+{
+	int *first_status = (int *)user_data;
+	int status = net_if_down(iface);
+
+	if (status == 0) {
+		return;
+	}
+
+	if (*first_status == 0) {
+		*first_status = status;
+	}
+
+	NET_ERR("net_if_down failed for Iface %d (%p). ERR: %d",
+		net_if_get_by_iface(iface), iface, status);
+}
+
+static void conn_mgr_all_if_connect_cb(struct net_if *iface, void *user_data)
+{
+	int *first_status = (int *)user_data;
+	int status = 0;
+
+	/* First, take iface up if it isn't already */
+	if (!net_if_is_admin_up(iface)) {
+		conn_mgr_all_if_up_cb(iface, &status);
+	}
+
+	if (status != 0) {
+		if (*first_status == 0) {
+			*first_status = 0;
+		}
+		return;
+	}
+
+	/* We expect net_if_connect to be supported if the iface has a connectivity API at all.
+	 * Otherwise, the API would be incomplete, since the connect function is not optional.
+	 */
+	if (!net_if_supports_connectivity(iface)) {
+		return;
+	}
+
+	status = net_if_down(iface);
```

Comment:
This should be `net_if_connect`

Reply:
Yikes, good catch -- I'm glad unit tests for conn_mgr are the next thing on my agenda